### PR TITLE
Improving the coin search bar filter logic

### PR DESCRIPTION
**Pull request by anthotsang**
https://app.asana.com/0/683768070860536/854461086638935

### CODE REVIEW

File: app/models/coin.rb [hunk 1]
```diff
@@ -110,12 +110,12 @@ def summary
...
 
-    most_common_feed_source_name = self.most_common_feed_source.name
-    most_common_news_category_name = self.most_common_news_category.name
+    most_common_feed_source_name = self.most_common_feed_source&.name
+    most_common_news_category_name = self.most_common_news_category&.name
     result << %W[
       The most common news source covering #{self.name} is #{most_common_feed_source_name} and the
       most common news category is #{most_common_news_category_name}.
-    ]
+    ] if most_common_feed_source_name != nil and most_common_news_category_name != nil
```
Comment (hanchang) [inline app/models/coin.rb]: Thanks for the catch @anthotsang !

Jesse has an existing fix that's already deployed though, so let's roll this one back please to handle the merge conflict.

File: app/controllers/api/coins_controller.rb [hunk 2]
```diff
@@ -23,9 +23,17 @@ def search_by_params
     if params[:coinSlugs].present? 
       coins = Coin.where(slug: params[:coinSlugs])
     elsif params[:name].present?
-      coins = Coin.where('upper(concat(symbol,\' \', name)) like upper(?)', "%#{params[:name]}%")
-                  .order('length(name) asc')
-                  .limit(10)
+      coins = Coin.find_by_sql("
```
Comment (hanchang) [inline app/controllers/api/coins_controller.rb]: @anthotsang Just curious about how the query performance on this?

Comment (anthotsang) [reply app/controllers/api/coins_controller.rb]: Seems to be similar to the previous query. The EXPLAIN() for both has the same plan/costs.

File: app/models/coin.rb (hunk 1 above)
Comment (anthotsang) [reply app/models/coin.rb]: Reverted

**Pull request merged**